### ISSUE TITLE
It's not possible to upload zip archive created on Mac

### ISSUE THREAD

**Issue by chloegugli** [ow2-proactive/catalog#121]
There is a folder (name __MACOSX) inside Mac archives and the catalog tries to parse it as a workflow. As a result, we get an error "The object(s) couldn't be uploaded: It was not possible to parse an object: com.ctc.wstx.exc.WstxIOException: Invalid UTF-8 middle byte 0x0 (at char #99, byte #-1)" and no workflow is uploaded.

**Comment by chloegugli** [comment]
[Archive 3.zip](https://github.com/ow2-proactive/catalog/files/2148511/Archive.3.zip)
[YolloTest.zip](https://github.com/ow2-proactive/catalog/files/2148512/YolloTest.zip)

**Comment by ShatalovYaroslav** [comment]
It's true that zip should contain only files that will be extracted as catalog objects.
So catalog in current realization is not expecting to have folder inside zip.
The solutions are:
1) don't put folders during generating zip archive (use other library maybe on Mac)
2) inside catalog skip all folders inside zip archive.

**Comment by marcocast** [comment]
I tried to create a zip from my mac and I don't have the __MACOSX folder : 
[ciaos.zip](https://github.com/ow2-proactive/catalog/files/2148784/ciaos.zip)
which is strange as the mac SHOULD create that folder. 
Anyway a mac user can remove the folder with this command : 

zip -d your-archive.zip "__MACOSX*"

We can add this command in the documentation or in the tooltip in the portal maybe...

As a more robust solution we could opt for ignoring folders in the zip and only considering files.

**Comment by ShatalovYaroslav** [comment]
I agree as an improvement, on catalog service side we can skip the sub folders of archive.
But we can not handle the case if the required object files are inside subfolder.

**Comment by chloegugli** [comment]
Ignoring folders in the zip is a good solution. @marcocast Actually it seems that Mac users can't see this folder, it is hidden. You can see it if you open it in another OS.
